rfcs/text/Fusillade in the DCP.md

```diff
@@ -0,0 +1,105 @@
+
+## Shepherd
+***Leave this blank.** This role is assigned by DCP PM to guide the **Author(s)** through the RFC process.*
+
+*Recommended format for Shepherds:*
+
+ `[Name](mailto:username@example.com)`
+
+## Motivation
+
+Fusillade is responsible for providing a uniform authentication and authorization experiance across the DCP. As the DCP 
+moves forward with integrating fusillade into the DCP, processes needs to be put into place to prevent excessive 
+breaking of the system due to permission errors. By agreeing on a process for managing the permission in Fusillade, 
+errors relating to permissioning can be avoided, and will be easier to track down when they occur. 
+
+The goal of have an agreed process is to simplify the manage of Fusillade Roles, Groups, Users, Service Accounts across 
+the DCP.
+ 
+[fusillade:#226](https://github.com/HumanCellAtlas/fusillade/issues/226)
+
+### Glossary
+
+### User Stories
+
+1. As a component of the DCP, I would like to define permissions to limit access to specific resources.
+1. As a component of the DCP, I would like to define a group of users to manage the permission of a large groups of 
+users easily.
+1. As a component of the DCP, I would like to give permission to unauthenticated users.
+1. As a component of the DCP, I would like to use service account to programmatically manage my service.  
+
+## Scientific "guardrails" [optional]
+
+*Describe recommended or mandated review from HCA Science governance to ensure that the RFC addresses the needs of the scientific community.*
+
+## Detailed Design
+
+### As a component of the DCP, how do I control access to my resources?
+1. Define what actions and resources you'd like to control. Actions should take the form `{component}:{Action}`, for example to get a bundle from the DSS the action could be `dss:GetBundle`. The action is case sensitive, and camel case is suggested. Resources are what actions affect. The same action can be applied to multiple different resources. For the resource use the AWS ARN format. For example...
+
+1. Define the different types of users you need to support based on what actions and resource they will need permission for. These users types you define will be your roles. 
+
+1. Create a file **{role}.json** for each user type. Using the resource and actions previously define, write a policy in the AWS IAM policy format that express what each role can do. If it isn't explicitly stated, it is assumed the user with this role does not have permission. For the resource field, `*` can be used as a wild card in the same way it's used in AWS IAM policies.
+
+1. Run the script `??? -path "path to roles" -component "name of the component" -stage "current deployment stage" -secret "secretId of a service account in AWS" -service "path to service account credentials"` to add your roles to fusillade. The role that is added to fusillade will be named `{component}-{stage}-{role}`. The stage is not included for *prod* deployments.
```

Comment:
`??? -path "path to roles" -component "name of the component" -stage "current deployment stage" -secret "secretId of a service account in AWS" -service "path to service account credentials"`

Guessing this should include a `-role "role_name"` option if it's going to be named `{component}-{stage}-{role}`.  I think it would be useful to warn users when a role with that name already exists, and maybe add a `--force` option to overwrite existing roles.  Also, kind of a nit, but I think I prefer `--` options rather than `-` just to be consistent with the `data-store`.

Reply:
The path would be to a folder which contains all of the roles, so they can be updated all at once. It's probably better to do it one at a time so you know what role failed. I'll add `--` to the parameters, `--role`, change `--path` to `--file`. Error returned if role exists, and `--force` will overwrite the role if the roles exists.